DataFormats/TrackerCommon/src/LegacyTrackerTopology.cc

```diff
@@ -0,0 +1,233 @@
+#include "DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h"
+
+#include "FWCore/Concurrency/interface/Xerces.h"
+#include <xercesc/parsers/XercesDOMParser.hpp>
+#include <xercesc/sax/HandlerBase.hpp>
+#include <xercesc/dom/DOM.hpp>
+#include "FWCore/Utilities/interface/Exception.h"
+
+using namespace xercesc;
+
+namespace {
+  // copy into a std::string and clean up the C-style string
+  inline std::string xmlc_to_stdstring(char* cstr)
+  {
+    std::string str{cstr};
+    XMLString::release(&cstr);
+    return str;
+  }
+  // transcode, convert into a std::string and clean up the intermediate C-style string
+  inline std::string xmlc_to_stdstring(const XMLCh* xcstr)
+  {
+    char* cstr = XMLString::transcode(xcstr);
+    std::string str{cstr};
+    XMLString::release(&cstr);
+    return str;
+  }
+
+  // RAII release the XMLCh*
+  class auto_XMLString
+  {
+  public:
+    auto_XMLString(XMLCh* xch) : m_xch(xch) {}
+    XMLCh* get() const { return m_xch; }
+    ~auto_XMLString() { if ( m_xch ) { XMLString::release(&m_xch); } }
+    // avoid double release: make this class move-only (rule of 5)
```

Comment:
It used to be called the rule of three until C++11... "whenever any of the destructor, copy constructor, copy assignment operator, move constructor, and move assignment constructor is defined (i.e. non-default), they should all be defined", but at second thought it doesn't belong in a comment, so I took it out again.